core/src/main/java/com/digitalpebble/stormcrawler/protocol/httpclient/HttpProtocol.java

```diff
@@ -99,24 +107,42 @@ public void configure(final Config conf) {
         String proxyHost = ConfUtils.getString(conf, "http.proxy.host", null);
         int proxyPort = ConfUtils.getInt(conf, "http.proxy.port", 8080);
 
+        String proxyUser = ConfUtils.getString(conf, "http.proxy.user", null);
+        String proxyPass = ConfUtils.getString(conf, "http.proxy.pass", null);
+        String proxyAuthScheme = ConfUtils.getString(conf, "http.proxy.authscheme", AuthSchemes.BASIC);
+        List<String> authSchemes = Lists.newArrayList();
+
         boolean useProxy = proxyHost != null && proxyHost.length() > 0;
 
         // use a proxy?
         if (useProxy) {
+
+            if (StringUtils.isNotBlank(proxyUser) && StringUtils.isNotBlank(proxyPass)) {
+                if (!SUPPORTED_AUTH_SCHEMES.contains(proxyAuthScheme)) {
+                    LOG.error("Configured proxy auth scheme: {} is not supported. Proxy authentication will not be used", proxyAuthScheme);
+                } else {
+                    authSchemes.add(proxyAuthScheme);
+                    BasicCredentialsProvider basicAuthCreds = new BasicCredentialsProvider();
+                    basicAuthCreds.setCredentials(new AuthScope(proxyHost, proxyPort),
+                            new UsernamePasswordCredentials(proxyUser, proxyPass));
+                    builder.setDefaultCredentialsProvider(basicAuthCreds);
+                }
+            }
+
             HttpHost proxy = new HttpHost(proxyHost, proxyPort);
-            DefaultProxyRoutePlanner routePlanner = new DefaultProxyRoutePlanner(
-                    proxy);
+            DefaultProxyRoutePlanner routePlanner = new DefaultProxyRoutePlanner(proxy);
             builder.setRoutePlanner(routePlanner);
         }
 
         int timeout = ConfUtils.getInt(conf, "http.timeout", 10000);
 
-        Builder requestConfigBuilder = RequestConfig.custom();
-        requestConfigBuilder.setSocketTimeout(timeout);
-        requestConfigBuilder.setConnectTimeout(timeout);
-        requestConfigBuilder.setConnectionRequestTimeout(timeout);
-        requestConfigBuilder.setCookieSpec(CookieSpecs.STANDARD);
-        requestConfig = requestConfigBuilder.build();
+        requestConfig = RequestConfig.custom()
+                .setSocketTimeout(timeout)
+                .setConnectTimeout(timeout)
+                .setConnectionRequestTimeout(timeout)
+                .setCookieSpec(CookieSpecs.STANDARD)
+                .setProxyPreferredAuthSchemes(authSchemes)
```

Comment:
what happens if authSchemes is empty? line 122 above

as suggested - let's have just the basic scheme. If so, what value would be used if we did not call this method?